ntbs-service/Pages/Index.cshtml

```diff
@@ -5,86 +5,158 @@
     ViewData["Title"] = "Home page";
     Layout = "./Shared/_NoForm_Layout.cshtml";
 }
+<tb-service-filtered-alerts inline-template>
+    <div>
+        <div class="flex-container">
+            <h2 class="alerts-heading"> Alerts </h2>
+            <label nhs-label-type="Standard" for="tb-service-filter-dropdown" classes="tb-service-alerts-filter-label">
+                Filter by TB Service
+            </label>
+            <select id="tb-service-filter-dropdown" nhs-select-type="Standard" ref="tbService" asp-items="Model.TbServices" v-on:change="updateTable" classes="tb-service-alerts-filter-dropdown">
+                <option> Show all </option>
+            </select>
+        </div>
+
+        <nhs-table class="notifications-table">
+            <nhs-table-head>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].NotificationId)
+                </nhs-table-item>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].FormattedCreationDate)
+                </nhs-table-item>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].AlertType)
+                </nhs-table-item>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].CaseManagerFullName)
+                </nhs-table-item>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].HospitalName)
+                </nhs-table-item>
+                <nhs-table-item>
+                    @Html.DisplayNameFor(model => model.Alerts[0].DismissLink)
+                </nhs-table-item>
+            </nhs-table-head>
+            <nhs-table-body>
+                @foreach (var item in Model.Alerts) 
+                {
+                    <nhs-table-body-row ref="@item.TbServiceCode" v-if="TbServiceCode == '@item.TbServiceCode' || TbServiceCode == 'Show all'">
+                        <nhs-table-item>
+                            @if(item.NotificationId != null) 
+                            {
+                                <a href="@RouteHelper.GetNotificationPath(NotificationSubPaths.Overview, item.NotificationId ?? 1)" class="govuk-link govuk-link--no-visited-state">
```

Comment:
Also I don't think we should have default id here? We should either have the NotificationId, and then no need to default it to 1, or not have it, in which case we shouldn't be showing a link here?

Reply:
the function accepts an int instead of int? and it doesn't seem smart enough NotificationId is not null from the if loop previous to this statement - what is the best way to fix this?

Reply:
oh, I see - you are already coniditioning on it.
Hmm, does saying `@RouteHelper.GetNotificationPath(NotificationSubPaths.Overview, (int)item.NotificationId)` help?
Or perhaps even better `@RouteHelper.GetNotificationPath(NotificationSubPaths.Overview, item.NotificationId.GetValueOrDefault())` (https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.getvalueordefault?view=netframework-4.8#System_Nullable_1_GetValueOrDefault)